src/fields/field_fe.cc

```diff
@@ -158,12 +161,74 @@ VectorMPI FieldFE<spacedim, Value>::set_fe_data(std::shared_ptr<DOFHandlerMultiD
 	init_data.data_vec = data_vec_;
 	init_data.ndofs = ndofs;
 	init_data.n_comp = this->n_comp();
-	init_data.comp_index = component_index;
+	init_data.mixed_fe = this->fe_;
 
 	// initialize value handler objects
+	init_data.range_begin = this->fe_item_[0].range_begin_;
+	init_data.range_end = this->fe_item_[0].range_end_;
 	value_handler0_.initialize(init_data);
+	init_data.range_begin = this->fe_item_[1].range_begin_;
+	init_data.range_end = this->fe_item_[1].range_end_;
 	value_handler1_.initialize(init_data);
+	init_data.range_begin = this->fe_item_[2].range_begin_;
+	init_data.range_end = this->fe_item_[2].range_end_;
 	value_handler2_.initialize(init_data);
+	init_data.range_begin = this->fe_item_[3].range_begin_;
+	init_data.range_end = this->fe_item_[3].range_end_;
+	value_handler3_.initialize(init_data);
+
+	// set discretization
+	discretization_ = OutputTime::DiscreteSpace::UNDEFINED;
+	interpolation_ = DataInterpolation::equivalent_msh;
+
+	return data_vec_;
+}
+
+
+template <int spacedim, class Value>
+VectorMPI FieldFE<spacedim, Value>::set_fe_system_data(std::shared_ptr<DOFHandlerMultiDim> dh,
```

Comment:
Merge separated methods into one,move block to last parameter, use default undef int value.